### PR TITLE
Extended track metadata [aoide Prerequisite]

### PR DESCRIPTION
**Pull request by uklotzde**
**Preamble:** *This is actually not a mandatory requirement for aoide integration. But I have written all the export function to include all these new properties and it would be a pity to dismantle this.*

**Bugfix:** Includes a fix to handle the new ID3v2 frame "GRP1" that has been introduced by iTunes: https://bugs.launchpad.net/mixxx/+bug/1844919. The code for this fix is enabled even without definiting `__EXTRA_METADATA__`.

Adds support for importing and exporting various popular file tags. No database changes, i.e. the new tags are transient and only available if freshly imported from a file. They are discarded when the `Track` object goes out of scope. Update: Hidden behind (yet) unused define `__EXTRA_METADATA__`.

The existing functionality should not be affected. Therefore the export of empty/missing values must be disabled to preserve existing file tags!! This special case handling is mandatory even if we would add those new tags to our database. Without reimporting all files the new fields would contain a NULL value initially.

We can decide how to actually handle the migration when we add some of those fields to our database schema. Hint: For the *total tracks* field we use a special placeholder in the database to distinguish unknown from missing values.

The handling of work/movement tags is tricky because Apple decided to remap existing tags.

### CODE REVIEW

File: src/track/trackmetadatataglib.cpp [hunk 1]
```diff
@@ -1691,6 +1916,15 @@ void importTrackMetadataFromMP4Tag(TrackMetadata* pTrackMetadata, const TagLib::
         pTrackMetadata->refTrackInfo().setGrouping(grouping);
     }
 
+    QString work;
```
Comment (daschuer) [inline src/track/trackmetadatataglib.cpp]: Do you have here an external reference link as well. This would be a nice addition.

**Review by daschuer — CHANGES_REQUESTED** [review]
Thank you for the good additions. 

What will happen in master after merge?
It will read some more unused tags from the files? 

If yes, we should #ifdef them out, to not slow down the library scanner and risk crashes because of yet unknown issues when reading or writing them.  

What do you think?

File: src/track/trackmetadatataglib.cpp [hunk 2]
```diff
@@ -1691,6 +1916,15 @@ void importTrackMetadataFromMP4Tag(TrackMetadata* pTrackMetadata, const TagLib::
...
+    if (readMP4Atom(tag, "\251wrk", &work)) {
```
Comment (daschuer) [inline src/track/trackmetadatataglib.cpp]: I have been stumbled over \251. Is that a kind of legacy? Do you know the origin?  From the latin-1 code page it is the ©
I think we should change it to the \xa9 notation everywhere. I have found it in this notation in other references and for me the octal decoding is kind of untypical unfamiliar. Github finds 15 vs 126 for mvn

File: src/track/trackinfo.h [hunk 3]
```diff
@@ -21,22 +21,29 @@ class TrackInfo final {
...
     PROPERTY_SET_BYVAL_GET_BYREF(QString,    key,                  Key)
     PROPERTY_SET_BYVAL_GET_BYREF(QString,    isrc,                 ISRC)
     PROPERTY_SET_BYVAL_GET_BYREF(QString,    language,             Language)
     PROPERTY_SET_BYVAL_GET_BYREF(QString,    lyricist,             Lyricist)
     PROPERTY_SET_BYVAL_GET_BYREF(QString,    mood,                 Mood)
+    PROPERTY_SET_BYVAL_GET_BYREF(QString,    movement,             Movement)
     PROPERTY_SET_BYVAL_GET_BYREF(QUuid,      musicBrainzArtistId,  MusicBrainzArtistId)
+    PROPERTY_SET_BYVAL_GET_BYREF(QUuid,      musicBrainzRecordingId, MusicBrainzRecordingId)
     PROPERTY_SET_BYVAL_GET_BYREF(QUuid,      musicBrainzReleaseId, MusicBrainzReleaseId)
-    PROPERTY_SET_BYVAL_GET_BYREF(QString,    recordLabel,          RecordLabel)
```
[the remark this answers is not in the document]
Comment (uklotzde) [reply src/track/trackinfo.h]: Commit message: "Add/move release-related properties to AlbumInfo"

File: src/track/trackmetadatataglib.cpp (hunk 1 above)
Comment (uklotzde) [reply src/track/trackmetadatataglib.cpp]: Added link on top of file to avoid repetition

**Comment by uklotzde** [comment]
> Thank you for the good additions.
> 
> What will happen in master after merge?
> It will read some more unused tags from the files?
> 
> If yes, we should #ifdef them out, to not slow down the library scanner and risk crashes because of yet unknown issues when reading or writing them.
> 
> What do you think?

Now hidden behind feature flag.

**Comment by uklotzde** [comment]
src/track/trackmetadatataglib.cpp:1945:26: warning: hex escape sequence out of range
"\xA9day"

I'll revert the commit and stick to 3-digit octals.

**Comment by uklotzde** [comment]
Fixes https://bugs.launchpad.net/mixxx/+bug/1844919

**Comment by daschuer** [comment]
LGTM, Thank you.

**Pull request merged**